src/daemon.rs

```diff
@@ -113,11 +113,27 @@ struct NetworkInfo {
 }
 
 #[derive(Serialize, Deserialize, Debug)]
-struct IndexInfo {
+struct TxIndexInfo {
     synced: bool,
     best_block_height: usize,
 }
 
+#[derive(Deserialize, Debug)]
+#[serde(rename_all = "lowercase")]
+enum IndexInfo {
+    TxIndex(TxIndexInfo),
+}
+
+enum TxIndexState {
+    /// Didn't check yet, re-check next time
+    Unknown,
+    /// assume unavailable but re-check after a minute since `last_check`
+    Syncing { last_check: std::time::Instant, },
+    /// Available and synced - do not pass blockhash
+    Synced,
+    /// `txindex=0` not configured - pass blockhash and don't retry check
+    Unavailable,
+}
```

Comment:
This is partial change, right?

Reply:
Yeah, still thinking about how using it now, quite difficult as you already told x)

Reply:
Just realized I forgot about `#[derive(Copy, Clone)]` it will not solve the problem but avoid annoyance later. :)